webapp/channels/src/packages/mattermost-redux/src/reducers/entities/files.ts

```diff
@@ -188,9 +188,96 @@ function filePublicLink(state: {link: string} = {link: ''}, action: GenericActio
     }
 }
 
+function filePreviews(state: Record<string, FilePreviewInfo[]> = {}, action: GenericAction) {
+    switch (action.type) {
+    case FileTypes.RECEIVED_FILE_PREVIEWS:
+        return {
+            ...state,
+            [action.pendingPostId]: action.data,
+        };
+
+    case FileTypes.UPDATE_FILE_UPLOAD_PROGRESS: {
+        const nextState = {...state};
+        const entries = Object.entries(nextState);
+        for (let entryIndex = 0; entryIndex < entries.length; entryIndex++) {
+            const entry = entries[entryIndex];
+            const filePreviewInfoIndex = entry[1].findIndex((f) => f.clientId === action.data.clientId);
+            if (filePreviewInfoIndex === -1) {
+                continue;
+            }
+            const pendingPostId = entry[0];
+            const filePreviewInfo = nextState[pendingPostId][filePreviewInfoIndex];
+            nextState[pendingPostId] = nextState[pendingPostId].map((f) => {
+                if (f.clientId !== filePreviewInfo.clientId) {
+                    return f;
+                }
+                return {
+                    ...f,
+                    percent: action.data.percent,
+                };
+            });
+            break;
+        }
+        return nextState;
+    }
+
+    case FileTypes.REMOVE_FILE_PREVIEWS: {
```

Comment:
Instead of needing a separate action for this, could we actually listen for a `RECEIVED_NEW_POST` here? If that post's `pending_post_id` is set, we can see if that ID is a key in `state` and then, if that's found, remove it